### PR TITLE
feat: Make `no_color` available as a project setting

### LINKED ISSUE #6940: refactor: Consider making `no_color` a project-level setting
### Feature scope

CLI (options, error messages, logging, etc.)

### Description

and documenting it in https://docs.meltano.com/reference/settings.

One potential caveat is that project settings can be set in places other that the environment, like `meltano.yml` and `.env`.

This would make it hard to configure things like the console log formatter like

```yaml
# https://docs.meltano.com/guide/logging
version: 1
disable_existing_loggers: false

formatters:
  structured_plain: # log format for structured plain text logs without colored output
    (): meltano.core.logging.console_log_formatter
    colors: False # also disables `rich` traceback formatting
  structured_colored: # log format for structured plain text logs WITH colored output
    (): meltano.core.logging.console_log_formatter
    colors: True # also enables traceback formatting with `rich`
```

which is (and should prob remain!) unaware of the Meltano project and just look for `NO_COLOR` directly in the global environment.

_From conversation with @tayloramurphy._

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes https://github.com/meltano/meltano/issues/6940

### CODE REVIEW

**Review by tayloramurphy — APPROVED** [review]

File: docs/src/_reference/settings.md [hunk 1]
```diff
@@ -419,6 +419,22 @@ root:
...
+Disable colored output on the command line. See [`no_color.org`](https://no-color.org/) for more information.
+
+#### How to use
+
+```bash
+meltano config meltano set cli no_color debug
+
+export MELTANO_CLI_NO_COLOR=true
+export NO_COLOR=true
+```
```
Comment (WillDaSilva) [inline docs/src/_reference/settings.md:436]: You may want to indicate that only one of these 3 command need be run.

**Review by WillDaSilva — APPROVED** [review]

**Review by aaronsteers — CHANGES_REQUESTED** [review]
Requesting we hold on this. Converting things to project settings has sometimes created new issues. I'd want to better understand any potential impacts versus benefits before moving forward.

**Comment by tayloramurphy** [comment]
@WillDaSilva @edgarrmondragon how are we feeling about this PR?

**Comment by edgarrmondragon** [comment]
> @WillDaSilva @edgarrmondragon how are we feeling about this PR?

@tayloramurphy It seems tests are failing, probably related to changes done in the interim to how project and settings are managed under the hood. It's probably take me a full day to figure out what's changed and update the PR.

**Comment by edgarrmondragon** [comment]
Closing this to keep things clean. If we reprioritize, we can start  a clean PR.

**Pull request closed without merging**